### DISCUSSION TITLE
How to Change id to slug ?

### DISCUSSION THREAD

**Discussion by bdloner** [strapi/strapi#6137] [Legacy Topics]
![image](https://user-images.githubusercontent.com/15920037/81470535-6eda7300-9215-11ea-8b55-31fbfbb20d58.png)

How can I change the id to slug?

At this time, the strapi backend has already been installed on Slugify.

![image](https://user-images.githubusercontent.com/15920037/81470590-ac3f0080-9215-11ea-9187-72dbb12b5ea8.png)

But I can't change route on the Nuxt.js

Thanks for the question area.

**Comment by robertpiosik** [comment]
@bdloner You should use a `UID` field for this purpose and query your endpoint like this: `localhost:8080/articles?uid=uid-field-value`. UID has unique index so retrieving data by this field is as efficient as it would be with an ID.

**Reply by bdloner** [reply]
Can I use slug instead because it's good for SEO?

**Reply by robertpiosik** [reply]
Hmm, how does it matter for you, as under this path you get `json` data? Try to make correct slug navigation on your frontend application level and make api calls using a `uid` field param.

**Reply by bdloner** [reply]
![image](https://user-images.githubusercontent.com/15920037/81775315-d1ce5180-9516-11ea-8e98-0cd269a182b6.png)

now i can change id to slug success when i leftclick on post but failed when i new tabs, WHY?

![image](https://user-images.githubusercontent.com/15920037/81775497-25d93600-9517-11ea-93da-43e5b3064603.png)

**Reply by robertpiosik** [reply]
Your issue is related with the frontend layer of your application. Inspect your api routes in postman and try to figure out if data you expect to get is actually there.

**Comment by lauriejim** [comment]
Hello! You will have to customize the controller of your Content Type by using this concept - [concept](https://strapi.io/documentation/3.0.0-beta.x/concepts/controllers.html#core-controllers)

Then in your findOne function you will have to replace
```diff
- const entity = await strapi.services.restaurant.findOne({ id });
+ const entity = await strapi.services.restaurant.findOne({ slug: id });
```
